crates/ruff_db/src/system/memory_fs.rs

```diff
@@ -207,7 +207,9 @@ impl MemoryFileSystem {
 
         let normalized = self.normalize_path(path.as_ref());
 
-        get_or_create_file(&mut by_path, &normalized)?.content = content.to_string();
+        let file = get_or_create_file(&mut by_path, &normalized)?;
+        file.content = content.to_string();
+        file.last_modified = FileTime::now();
```

Comment:
Another bug haha! the memory file system never updated the last modified timestamp and the invalidation only worked because salsa never compared if the input actually changed